downstairs/src/extent_inner_raw_v2.rs

```diff
@@ -0,0 +1,1058 @@
+                        CrucibleError::IoError(format!(
+                            "call to `zfs` failed: {e}"
+                        ))
+                    })?;
+                    let mut err = vec![];
+                    p.stderr.unwrap().read_to_end(&mut err).map_err(|e| {
+                        CrucibleError::IoError(format!(
+                            "failed to read stderr from `zfs`: {e:?}"
+                        ))
+                    })?;
+                    let err = std::str::from_utf8(&err).map_err(|e| {
+                        CrucibleError::IoError(format!(
+                            "zfs returned invalid UTF-8 string: {err:?} ({e})"
+                        ))
+                    })?;
+                    if err.contains("not a ZFS filesystem") {
+                        DEFAULT_ZFS_RECORDSIZE
+                    } else {
+                        let mut out = vec![];
+                        p.stdout.unwrap().read_to_end(&mut out).map_err(
+                            |e| {
+                                CrucibleError::IoError(format!(
+                                    "failed to read stdout from `zfs`: {e:?}, \
+                                     stderr: {err}"
+                                ))
+                            },
+                        )?;
+                        let out = std::str::from_utf8(&out).map_err(|e| {
+                            CrucibleError::IoError(format!(
+                                "zfs returned invalid UTF-8 string: {out:?} \
+                                 ({e}), stderr: {err}"
+                            ))
+                        })?;
+                        out.trim().parse::<u64>().map_err(|e| {
+                            CrucibleError::IoError(format!(
+                                "zfs returned non-integer for recordsize: \
+                                  {out:?} ({e}), stderr: {err}"
+                            ))
+                        })?
+                    }
+                }
+                Err(e) if e.kind() == std::io::ErrorKind::NotFound => {
+                    // If the `zfs` executable isn't present, then we're
+                    // presumably on a non-ZFS filesystem and will use a default
+                    // recordsize
+                    DEFAULT_ZFS_RECORDSIZE
+                }
+                Err(e) => {
+                    return Err(CrucibleError::IoError(format!(
+                        "could not call `zfs` executable: {e:?}"
+                    )))
+                }
+            }
+        };
+        Ok(recordsize)
+    }
+}
+
+/// Data structure that implements the on-disk layout of a raw extent file
+struct RawLayout {
+    extent_size: Block,
+    recordsize: u64,
+}
+
+impl std::fmt::Debug for RawLayout {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("RawLayout")
+            .field("extent_size", &self.extent_size)
```

Comment:
Why only this?

Reply:
This was copy-pasta from `extent_inner_raw.rs`, but I'm not sure why they're not just deriving Debug.

Fixed in 5b2a9425ea09 for both